OAuth/ResourceOwner/AbstractResourceOwner.php

```diff
@@ -342,6 +352,7 @@ protected function configureOptions(OptionsResolver $resolver)
 
         $resolver->setDefaults([
             'scope' => null,
+            'state' => null,
```

Comment:
I was seeing this recently and was unsure why this option was added. Seeing as these options are configured at build time, whatever you configure as state parameter will not change, right? This state doesn't change, so what would be the use case here?

Reply:
@paulinevos The state param can be passed in multiple ways, we are using it from query params that are sent to the resource owner, then after authentication sent back to in the response. Setting it to `null` by default will just generate a random hash that is passed to the RO then back, should you choose to use CSRF protection.

Reply:
Right, but when added directly to the URI I understand the use case, but seeing as these options will remain the same from build I don't understand how that manages state? Because these options never change, you could just fetch them from the options array any time you wish instead of passing them into a query parameter on a roundtrip.